eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/GovSpec.hs

```
@@ -250,10 +251,72 @@ pparamUpdateSpec =
 
 proposalsWithVotingSpec ::
   forall era.
-  ConwayEraImp era =>
+  ( ConwayEraImp era
+  , InjectRuleFailure "LEDGER" ConwayGovPredFailure era
+  ) =>
   SpecWith (ImpTestState era)
 proposalsWithVotingSpec =
   describe "Proposals" $ do
+    describe "Failures expected after bootstrap phase" $ do
+      it "Fails predicate when treasury withdrawal has nonexistent return address" $ do
+        policy <- getGovPolicy
+        protVer <- getProtVer
+        registeredRewardAccount <- registerRewardAccount
+        netId <- use (impGlobalsL . to networkId)
+        unregisteredRewardAccount <- RewardAccount netId . KeyHashObj <$> freshKeyHash
+        deposit <- getsNES $ nesEsL . curPParamsEpochStateL . ppGovActionDepositL
+        anchor <- arbitrary
+        let mkTreasuryWithdrawal rewardAccount =
+              ProposalProcedure
+                { pProcDeposit = deposit
+                , pProcReturnAddr = registeredRewardAccount
+                , pProcGovAction =
+                    TreasuryWithdrawals (Map.singleton rewardAccount $ Coin 1_000) policy
+                , pProcAnchor = anchor
+                }
+            validTreasuryWithdrawal = mkTreasuryWithdrawal registeredRewardAccount
+            invalidTreasuryWithdrawal = mkTreasuryWithdrawal unregisteredRewardAccount
+        if HF.bootstrapPhase protVer
```

The function in which these tests are included  (proposalsWithVotingSpec), is only called in `spec` and not in `relevantDuringBootstrapSpec`, so as things are now, it makes no sense to add this check here, because the flag will will always be False. 
So, either you check only the post-bootstrap scenario (so without any `if HF.bootstrapPhase`), or you could move the test in a place that gets called from `relevantDuringBootstrapSpec`,  and that will indeed verify the behavior for both protocol versions. 
 However, in this particular case, since withdrawals are not actually allowed during bootstrap, not sure if  it makes sense to have it  for both versions - i guess it depends on whether we want to add the predicate failure for the bootstrap phase as well.

---

I want the tests to run for both protocol versions, ideally, so that we are certain that not only are treasury withdrawals not allowed during the bootstrap phase, but also, the predicate failure is not reported for invalid return addresses. These failures should only be reported after the bootstrap phase so we don't break node-client protocol. When I was implementing these tests they did run for both protocol versions. Has something changed recently that makes these run only under `relevantDuringBootstrapSpec` now?

---

It works like this: 
 - Gov. relevantDuringBootstrapSpec  is called only for protocol version 9 
 - Gov.spec (which itself calls relevantDuringBootstrapSpec) is called for version 10 
 
This means that  in order to have something be called for both version 9 and 10, we should include it in `relevantDuringBootstrapSpec` . Everything that is called in `spec` besides `relevantDuringBootstrapSpec` will only be called for 10 :

```
spec = do
  relevantDuringBootstrapSpec    -- called separately for 9  in `Imp` module 
  constitutionSpec   -- only 10
  proposalsWithVotingSpec  -- only 10. 
  votingSpec --..etc 
  policySpec
  predicateFailuresSpec
  unknownCostModelsSpec
```

Because you added the tests to `proposalsWithVotingSpec` , they are only invoked with 10. 
So I think it would be good to either add them to a function already called in `relevantDuringBootstrapSpec` or create a new one and call it from  `relevantDuringBootstrapSpec` 

Hope this makes sense, let me know if not.

---

Well, my bad, I see now. I have moved the tests to the appropriate parts of `relevantDuringBootstrapSpec`.